src/openrct2/ride/Ride.cpp

```diff
@@ -5983,3 +5983,14 @@ ResultWithMessage Ride::ChangeStatusCreateVehicles(bool isApplying, const Coords
 
     return { true };
 }
+
+uint8_t Ride::GetEntranceStyle() const
+{
+    const auto* stationObject = GetStationObject();
+    if (stationObject == nullptr)
+        return RCT12_STATION_STYLE_PLAIN;
+
+    const auto objectName = stationObject->GetIdentifier();
+
+    return GetStationStyleFromIdentifier(objectName);
```

Comment:
```suggestion
    if (const auto* stationObject = GetStationObject() != nullptr)
    {
        const auto objectName = stationObject->GetIdentifier();
        return GetStationStyleFromIdentifier(objectName);
    }
    else
    {
        return RCT12_STATION_STYLE_PLAIN;
    }
```

Reply:
This didn't compile for me but I did something similar

Reply:
This is simply incorrect. Station objects have a flag that you can check. It should never check RCT2 ID’s for anything. The function was defined in TrackDesign.cpp for a **very** good reason.